mapie/conformity_scores/residual_conformity_scores.py

```diff
@@ -132,3 +141,275 @@ def get_estimation_distribution(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        y_pred: ArrayLike
+    ) -> Tuple[NDArray, NDArray, NDArray, RegressorMixin]:
+        """
+        Checks all the parameters of the class. Raises an error if the
+        parameter are not well defined.
+
+        Parameters
+        ----------
+        X: ArrayLike
+            Observed values.
+
+        y: ArrayLike
+            Target values.
+
+        y_pred: ArrayLike
+            Predicted targets.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray, NDArray, RegressorMixin]
+        Well initiated and typed :
+            - X
+            - y
+            - y_pred
+            - residual_estimator
+        """
+        residual_estimator = self._check_estimator(
+            self.residual_estimator
+        )
+        check_random_state(self.random_state)
+        if self.split_size is None:
+            self.split_size = 0.5
+        X, y, y_pred = indexable(X, y, y_pred)
+        X = np.array(X)
+        y = np.array(y)
+        y_pred = np.array(y_pred)
+        return X, y, y_pred, residual_estimator
+
+    def _fit_residual_estimator(
+        self,
+        residual_estimator_: RegressorMixin,
+        X: NDArray,
+        y: NDArray,
+        y_pred: NDArray,
+        calres_indexes: NDArray
+    ) -> Tuple[NDArray, NDArray]:
+        """
+        Fit the residual estimator and returns the indexes used for the
+        training of the base estimator and those needed for the calibration.
+
+        Parameters
+        ----------
+        X : NDArray
+            All the observed values used in the general fit.
+
+        y : NDArray
+            All the observed targets used in the general fit.
+
+        y_pred : NDArray
+            Predicted targets.
+
+        calres_indexes : NDArray
+            Indexes used for the training of the estimaot and the calibration.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray]
+            - indexes needed for the calibration.
+            - indexes used for the training of the base estimator.
+        """
+        (X_res_indexes,
+         X_cal_indexes,
+         y_res_indexes,
+         y_cal_indexes) = train_test_split(
+            calres_indexes,
+            calres_indexes,
+            test_size=self.split_size,
+            random_state=self.random_state,
+        )
+
+        residuals = np.abs(np.subtract(
+            y[y_res_indexes],
+            y_pred[y_res_indexes]
+        ))
+        residual_estimator_targets = np.log(np.maximum(
+            residuals,
+            np.full(residuals.shape, self.eps)
+        ))
+
+        residual_estimator_ = residual_estimator_.fit(
+            X[X_res_indexes],
+            residual_estimator_targets
+        )
+
+        cal_index = X_cal_indexes
+        train_index = list(set(np.arange(y_pred.shape[0])) - set(cal_index))
+
+        self.residual_estimator_ = residual_estimator_
+
+        return cal_index, np.array(train_index)
```

Comment:
SUGGESTION: Can you adapt your code to handle variables in a simpler way? This is a suggestion to test and adapt:
```suggestion
        X, y, y_pred, residual_estimator_ = self._check_parameters(
            X, y, y_pred
        )
        
        residuals = np.abs(np.subtract(y, y_pred))
        targets = np.log(np.maximum(residuals, self.eps))

        self.residual_estimator_ = residual_estimator_.fit(X, targets)

        return self
```
It's a way of doing the same thing as any scikit-learn estimator (as `MapieRegressor` does, for example). This also allows you to better separate the split from the train. I suggest renaming it `fit` with signature `fit(X, y, y_pred)`.

Reply:
The fit does not handle `nan` values